bridges/snowbridge/primitives/inbound-queue/src/v2/converter.rs

```diff
@@ -200,25 +215,36 @@ where
 			],
 		);
 
+		// If the claimer is an AccountId32 on AH, use it to refund excess fees.
+		// Otherwise, use the bridge owner.
+		let claimer_account = match claimer.unpack() {
+			(0, [AccountId32 { id, .. }]) => *id,
+			_ => bridge_owner,
+		};
+
 		match network {
 			super::message::Network::Polkadot => Ok(Self::make_create_asset_xcm_for_polkadot(
 				create_call_index,
 				asset_id,
 				bridge_owner,
 				dot_fee,
 				eth_asset,
+				claimer_account,
 			)),
 		}
 	}
 
+	/// Construct the asset creation XCM for the Polkdot network.
 	fn make_create_asset_xcm_for_polkadot(
 		create_call_index: [u8; 2],
 		asset_id: Location,
 		bridge_owner: [u8; 32],
 		dot_fee_asset: xcm::prelude::Asset,
 		eth_asset: xcm::prelude::Asset,
+		claimer_account: [u8; 32],
 	) -> Xcm<()> {
 		vec![
+			RefundSurplus,
```

Comment:
Isn't this supposed to be after the transact?

Reply:
https://github.com/vgeddes/polkadot-sdk/pull/32/commits/7ab4934288fe54be67127a87b90c6f99249221ef